tempest/api/workloadmgr/multiattach_volumes/test_multiattach_volumes.py

```diff
@@ -0,0 +1,337 @@
+            LOG.debug(
+                f"md5sums_before_incr: {md5sums_before_incr2} md5sums_before_incr_vol: {md5sums_before_incr2_vol}")
+
+            self.snapshot_id2 = self.workload_snapshot(self.wid, False)
+            self.wait_for_workload_tobe_available(self.wid)
+            self.snapshot_status = self.getSnapshotStatus(self.wid,
+                                                          self.snapshot_id2)
+            if (self.snapshot_status == "available"):
+                reporting.add_test_step("Create incremental snapshot", tvaultconf.PASS)
+                self.mount_path = self.get_mountpoint_path(
+                    tvaultconf.tvault_ip[0], tvaultconf.tvault_username,
+                    tvaultconf.tvault_password)
+                self.snapshot_found = self.check_snapshot_exist_on_backend(
+                    tvaultconf.tvault_ip[0], tvaultconf.tvault_username,
+                    tvaultconf.tvault_password, self.mount_path,
+                    self.wid, self.snapshot_id2)
+                LOG.debug(f"snapshot_found: {self.snapshot_found}")
+                if self.snapshot_found:
+                    reporting.add_test_step("Verify snapshot existence on " \
+                                            "target backend", tvaultconf.PASS)
+
+                    tests[2][1] = 1
+                    reporting.test_case_to_write()
+                else:
+                    raise Exception("Verify snapshot existence on target backend")
+            else:
+                raise Exception("Create incremental snapshot")
+
+
+            # selective restore
+            reporting.add_test_script(tests[3][0])
+            rest_details = {}
+            rest_details['rest_type'] = 'selective'
+            rest_details['volume_type'] = vol_type_name
+            rest_details['network_id'] = CONF.network.internal_network_id
+            rest_details['subnet_id'] = self.get_subnet_id(
+                CONF.network.internal_network_id)
+            rest_details['instances'] = {self.vm_id_1: self.volumes, self.vm_id_2: self.volumes}
+
+            payload = self.create_restore_json(rest_details)
+            # Trigger selective restore of full snapshot
+            self._selective_restore(payload,[fip[2],fip[3]],
+                                    [md5sums_before_full1, md5sums_before_full1_vol,
+                                     md5sums_before_full2, md5sums_before_full2_vol])
+
+            # Trigger selective restore of incremental snapshot
+            self._selective_restore(payload, [fip[4], fip[5]],
+                                    [md5sums_before_incr1, md5sums_before_incr1_vol,
+                                     md5sums_before_incr2,md5sums_before_incr2_vol],
+                                    False)
+            reporting.test_case_to_write()
+            tests[3][1] = 1
+
+
+
+            try:
+                self.detach_volume(self.vm_id_1, self.volume_id)
```

Comment:
Why vm_id_2 volumes are not detached ? Any specific reason ?

Reply:
because when Vm_id_1 volume is dettached, since its multiattach volume, it gets dettached from Vm 2 as well. So in VM 2 its not required.